### PR TITLE
v4-dev: NES Edition

### PR DESCRIPTION
**Pull request by mdo**
Replaces #39750 with fixes to make v4 docs run locally for me. Purpose of this is to establish a derivative of `v4-dev` that is explicitly for our NES related changes to the docs. No official release here is planned.

/cc @edezekiel

### CODE REVIEW

File: site/layouts/partials/announcement.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  Bootstrap 4 has reached end of life.
+  <a
+    href="https://getbootstrap.com/docs/5.3/migration/"
+    target="_blank"
+    rel="noopener"
+    >Upgrade to the latest</a
+  >
+  or consider
+  <a
+    href="https://www.herodevs.com/support/nes-bootstrap?utm_source=Bootstrap_site&utm_medium=Banner&utm_campaign=v4_eol"
```
Comment (julien-deramond) [inline site/layouts/partials/announcement.html:11]: I'm wondering if it wouldn't be better for the users to have a link to the new `end-of-life` page to have some information, and then they can decide whether or not they want to go to the HeroDevs website.

File: site/content/docs/4.6/end-of-life.md
Comment (julien-deramond) [inline site/content/docs/4.6/end-of-life.md:1]: More a global comment on this page. We are very specific by saying the latest version is version 5, we even go through the details of its content. This is nice for this transition v3/v4 -> v5. But in terms of maintenance, when we have a v6, v7, etc. how will it be handled?
Please be aware that this entire page will need to be modified for every version of Bootstrap in `gh-pages` after each release (except for dot releases).
The "What's Next" section is very well detailed, but how we will manage v6, v7, etc. in this section in the future from the v4 section, you'll write down a summary of every intermediate major versions?

File: site/content/docs/4.6/end-of-life.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+toc: true
+---
+
+Bootstrap 3 was released over a decade ago in August 2013, and Bootstrap 4 over six years ago in January 2018. Since then, millions of people have learned Bootstrap, prototyped with it, and shipped some amazing sites and apps with it. While Bootstrap 5 is continuing that journey with developers all over the world, there are still tons of those teams living on an older version of Bootstrap with limited to no support. Until now.
+
+Bootstrap 3 reached end of life July 24, 2019, followed by Bootstrap 4 on January 1, 2023. As such, neither version receives new features, updates, or security fixes. However, both are still available on all existing distribution channels (CDNs, package managers, GitHub, etc).
+
+We strongly recommend folks start out on or [upgrade](https://getbootstrap.com/docs/5.0/migration/) to the [latest version of Bootstrap (v5.x)](https://getbootstrap.com/), but for those who can’t upgrade just yet and have compliance or security requirements, we’re introducing [Never-Ending Support for Bootstrap 3 and 4 with HeroDevs](https://www.herodevs.com/support/nes-bootstrap?utm_source=Bootstrap_site&utm_medium=Banner&utm_campaign=v3and4_eol).
```
Comment (julien-deramond) [inline site/content/docs/4.6/end-of-life.md:14]:
```suggestion
We strongly recommend folks start out on or [upgrade](https://getbootstrap.com/docs/5.3/migration/) to the [latest version of Bootstrap (v5.x)](https://getbootstrap.com/), but for those who can’t upgrade just yet and have compliance or security requirements, we’re introducing [Never-Ending Support for Bootstrap 3 and 4 with HeroDevs](https://www.herodevs.com/support/nes-bootstrap?utm_source=Bootstrap_site&utm_medium=Banner&utm_campaign=v3and4_eol).
```

File: site/content/docs/4.6/end-of-life.md
Comment (mdo) [reply site/content/docs/4.6/end-of-life.md:1]: That's a good point. I can rev on the copy to be less specific so it lasts longer.

File: site/layouts/partials/announcement.html (hunk 1 above)
Comment (mdo) [reply site/layouts/partials/announcement.html:11]: @edezekiel Any thoughts on this from the HeroDevs side? I too am wondering about linking to this new page. We could add a third link to the banner, but I imagine that's some diminishing returns and fatigue to users.

File: site/content/docs/4.6/end-of-life.md
Comment (mdo) [reply site/content/docs/4.6/end-of-life.md:1]: Eh, actually it's not too bad re-reading. Replacing 5 with another number is easy enough, and I'd say we end up just dropping the v5-specific features stuff in the future. Plus down the line it'll be easier to modify an existing page as opposed to creating a new one.

Comment (aavorthmann) [reply site/content/docs/4.6/end-of-life.md:1]: So are we good on this part ? Do you need anything from the HeroDevs side ?

**Comment by mdo** [comment]
Deployed to production docs in #39786. We'll keep this branch around to work off of for future v4.x updates for NES, but otherwise is not needed and can close. Thanks all!

**Pull request closed without merging**